docs/tutorials/from-tabular-data-to-binned-data.ipynb

```diff
@@ -54,51 +67,29 @@
    },
    "source": [
     "We will use a file created by a simulation for a diamond sample using [McStas](http://www.mcstas.org/) and [Geant4](https://geant4.web.cern.ch/).\n",
-    "We can use [pandas.read_table](https://pandas.pydata.org/docs/reference/api/pandas.read_table.html) to load the table as a [pandas.Dataframe](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html).\n",
-    "`pandas.read_table` is much faster than [numpy.loadtxt](https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html) and also slightly more convenient:"
+    "We can use [scipp.io.load_csv](../generated/modules/scipp.io.csv.load_csv.rst) to load the file as a dataset.\n",
+    "The column headers of the file encode physical units in the form of `tof [ns]`.\n",
+    "The `'bracket'` header parser understands this format and extract the units."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "id": "b52a76b6-75ff-42b9-ad1f-843c7d4de5ba",
-   "metadata": {},
+   "metadata": {
+    "tags": []
+   },
    "outputs": [],
    "source": [
-    "import pandas as pd\n",
-    "\n",
     "filename = 'https://public.esss.dk/groups/scipp/scipp/1/data_dream_diamond.zip'\n",
-    "df = pd.read_table(filename)\n",
-    "df"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "id": "4b53eb28-08e9-4085-a4be-f96db3da5ca5",
-   "metadata": {},
-   "source": [
-    "[scipp.compat.from_pandas](https://scipp.github.io/generated/functions/scipp.compat.from_pandas.html) can convert the `pandas.Dataframe` to a `scipp.Dataset`.\n",
-    "The column names encode the physical units so we must extract them manually:"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "64ef53f4-c6d2-497c-af60-b25e28beee46",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "%matplotlib widget\n",
-    "import scipp as sc\n",
-    "import plopp as pp\n",
-    "\n",
-    "ds = sc.compat.from_pandas(df)\n",
-    "del ds.coords['row']  # we have no use for this row index\n",
-    "for key in list(ds):\n",
-    "    name, *remainder = key.split(' ')\n",
-    "    ds[name] = ds.pop(key)\n",
-    "    ds[name].unit = remainder[0][1:-1] if remainder else None\n",
-    "sc.table(ds[:10])"
+    "ds = sc.io.load_csv(\n",
+    "    filename,\n",
+    "    sep='\\t',\n",
```

Comment:
I remember a discussion on Friday, about making wrappers for all most common Pandas functions. Shouldn't we provide `load_table` then?

Reply:
I don't know which one is used more or what the difference even is.

Reply:
Not sure I see the value. Where would we put this? `io.table.load_table`? Or group by underlying implementation `io.pandas_io.{load_csv, load_table}`.

Reply:
If you do not see the value of `load_table`, then where is the value of `load_csv`?

`sc.io.load_csv` and `sc.io.load_table`?

Reply:
They are effectively the same function. So why have both? Also, we said that just because we add one function, we don't have to add all.

Reply:
`txt` could be anything. `csv` is more restricted to a table layout.
We could, *instead* of `load_csv`, have `load_table` if you think 'CSV' is not general enough. Though in my experience, people call anything with a table layout a csv file.

Reply:
Not sure about this one. If `read_table` is used a lot by pandas users, then we should maybe also provide it?
If it's just a thin wrapper around `load_csv`, it doesn't hurt too much to add it as it will just forward everything to `load_csv`.

The alternative is to name it `loadtxt` like Numpy, but use Pandas internally for speed. Then we would have a single function that would do both?

In any case, I think supporting the functionality of `read_csv` and `read_table` would be enough for now, and if people need `xml` and `json` they can use Pandas. If those turn out to be used a lot, then we can add wrappers in Scipp at a later stage?

Reply:
> If it's just a thin wrapper around load_csv, it doesn't hurt too much to add it as it will just forward everything to load_csv.

Implementation-wise, no, it doens't hurt much. But as you can see, there is confusion around why both functions exist. I think it is valuable to have a clear API that clearly tells you how to do something.

Reply:
So I would go with `loadtxt`? Not sure whether I agree that separator should be mandatory or not, or if it should behave like Numpy.

Reply:
I feel it would be confusing to have Pandas wrappers that are not named like Pandas functions, especially since we forward keyword arguments.

Reply:
But then we should maybe name them `read_csv` and `read_table` instead of `load`? I know we went with `load_hdf5`, so load seemed natural. But maybe we need to revisit?